twenty-first/src/shared_math/polynomial.rs

```diff
@@ -322,7 +328,14 @@ where
     }
 
     // domain: polynomial roots
-    pub fn fast_zerofier(domain: &[FF], primitive_root: &BFieldElement, root_order: usize) -> Self {
+    pub fn fast_zerofier<GF>(
+        domain: &[GF],
+        primitive_root: &BFieldElement,
+        root_order: usize,
+    ) -> Self
+    where
+        GF: FiniteField + Into<FF>,
```

Comment:
Adding a type parameter without a default is a breaking change, so `0.x+1.0`.

I'd probably mention that `GF: Into<FF>` means projecting `GF` into a superset in all of the cases of this PR.

Maybe that's a comment worth having at the top rather than at each `<GF: FiniteField>`.